### DISCUSSION TITLE
How to only deploy the assets to s3 without starting a cloudformation deploy

### DISCUSSION THREAD

**Discussion by jabalsad** [aws/aws-cdk#20780] [Q&A]
Hello,

I'm curious if it's possible for CDK to only upload the stack assets that are defined in <stack-name>.assets.json to S3 without initiating a cloudformation deploy?

**Comment by seetamraju** [comment]
Perhaps this is coming too late.

Instead of:
> `cdk deploy`

Use:
> `cdk deploy --method prepare-change-set`

1. All assets will be uploaded and ..
2. then, this cmd will create Change-Set (that is visible in the CloudFormation Console)
3. Finally, this cmd will get stuck at "Approval".
4. Cleanup: Delete that Change-Set (either manually or via aws-cli)

@jabalsad
